recipes-core/platform-onl-init/platform-onl-init_0.1.bb

```diff
@@ -0,0 +1,33 @@
+SUMMARY = "Platform Init service for ONL platforms"
+LICENSE = "MPL-2.0"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/MPL-2.0;md5=815ca599c9df247a0c7f619bab123dad"
+
+inherit systemd
+
+# This will actually be provided by the installer, but we need to pretend to
+# provide it so Yocto will be happy.
+RPROVIDES:${PN} += " u-boot-default-env"
+
+SRC_URI += " \
+    file://8v89307_init.sh \
+    file://platform-onl-init.service \
+    file://platform-onl-init.sh \
+    file://platform-arm-accton-as4610-30-r0-init.sh \
+    file://platform-arm-accton-as4610-54-r0-init.sh \
+    file://platform-x86-64-accton-as4630-54pe-r0-init.sh \
+    file://platform-x86-64-accton-as4630-54te-r0-init.sh \
+    file://platform-x86-64-accton-as5835-54x-r0-init.sh \
+    file://platform-x86-64-accton-as7726-32x-r0-init.sh \
+    file://platform-x86-64-cel-questone-2a-r0-init.sh \
+    file://platform-x86-64-delta-ag5648-r0-init.sh \
+    file://platform-x86-64-delta-ag7648-r0-init.sh \
+"
+
+do_install() {
+        install -d ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/platform-onl-init.service ${D}${systemd_system_unitdir}
+        install -d ${D}${bindir}
+        install -m 0755 ${WORKDIR}/*.sh ${D}${bindir}
```

Comment:
> I don't think ${bindir} is a good choice for these files. Users should not find a dozen or so files named something like platform-x86-64-accton-as5835-54x-r0-init.sh in their /usr/bin. Looking at the FHS and our current filesystem layout, I'd suggest creating and using a subdirectory under /usr/libexec, e.g. /usr/libexec/onl.

I see you point, though, but I would like to defer this to a later clean up. We might also want/need to not install platform kernel modules into `/lib/modules` because some vendors use the same module names for different platforms (with different contents!), and have the platform specific parts in a common directory. And to avoid moving around files multiple times I would do that once we decided for a good layout for that.

> On an unrelated note, install here results in the symlinks being resolved, meaning that the platform init files for the 4610 and 4630 are now present twice, as identical files with slightly different names. Not a big deal, but not pretty, either.

We can clean this up later, or even replace the symlink with a file with actual different content since we now know which platform this is.